src/mnode/mnode-pastel.cpp

```diff
@@ -603,14 +603,77 @@ bool common_validation(const T& ticket, bool preReg, const std::string& strTnxId
     if (depth > 0)
         return true;
     
-    std::string err;
     if (!pastelTicket->IsValid(false, ++depth)) {
-      throw std::runtime_error(strprintf("The %s ticket with this txid [%s] is invalid - %s", prevTicket, strTnxId, err));
+      throw std::runtime_error(strprintf("The %s ticket with this txid [%s] is invalid", prevTicket, strTnxId));
     }
     
     return true;
 }
 
+void trade_copy_validation(const std::string& nftTnxId, const std::vector<unsigned char>& signature) {
+  if (!masterNodeCtrl.masternodeSync.IsSynced()) {
+    throw std::runtime_error("Can not validate trade ticket as master node is not synced");
+  }
+
+  unsigned short totalCopies{0};
+
+  uint256 txid;
+  txid.SetHex(nftTnxId);
+  auto nftTicket = CPastelTicketProcessor::GetTicket(txid);
+  if (!nftTicket) {
+    throw std::runtime_error(strprintf(
+      "The NFT ticket with txid [%s] referred by this trade ticket is not in the blockchain", nftTnxId));
+  }
+  if (nftTicket->ID() == TicketID::Activate) {
```

Comment:
This code looks like copy&paste from CArtSellTicket::IsValid. 
Can we move it to a separate function and reuse in both places?
Duplicate code is so difficult to maintain, you change one place - but then forget to change all others.
Looks like the code in CArtSellTicket::IsValid is just getting some additional data from tickets.

Reply:
in CArtSellTicket::IsValid we have also additional checking for PastelID and copies are checked only with additional condition if (preReg || !ticketFound). So a few small changes but functions looks similar.

Reply:
I can remove this new function 'trade_copy_validation' as we ignore old sell/buy tickets and it is working correct now with latest fixes in the tests => so this function 'trade_copy_validation' in the end of CArtTradeTicket::IsValid is never fails. We can keep this function 'trade_copy_validation' just to be sure that all is correct and still available copies to sell. what do you think?

Reply:
you can leave it as it is.
We can think - maybe we can write later on some helper templates to get ticket and its properties (for different types of tickets).